src/main/java/com/microsoft/azure/servicebus/jms/ServiceBusJmsConnectionFactory.java

```diff
@@ -40,24 +35,32 @@ public class ServiceBusJmsConnectionFactory extends JNDIStorable implements Conn
     // JNDI property names
     private static final String CONNECTION_STRING_PROPERTY = "connectionString";
     private static final String CLIENT_ID_PROPERTY = "clientId";
-    private static final String AUDIENCE = "https://servicebus.azure.net/.default";
-    private static final String AUTHORITY = "https://login.microsoftonline.com/%s/";
     private static final int MaxCustomUserAgentLength = 128;
     
+    
+    // AAD TOKEN properties
+    private static final String AAD_TOKEN_USERNAME = "$jwt";
+    private boolean isMsi = false; 
+    private boolean isClientSecretCredential = false;
+    private boolean isDefaultAzureCredential = false;
+    private String tenantId; 
+    private String aadClientId; 
+    private String clientSecret;
+    private String aadToken;
+    
+    //SAS
+    private String sasKey;
+    private String sasKeyName;
+   
+    //Common properties
+    private boolean isAadAuthentication = false;
```

Comment:
To make it universal for all types of authorization you need to add only TokenCredential to the fields, and the actual object building should happen outside by end clients (It will depend on concrete TokenCredential implementation, you don`t need to handle it here). Look at the examples I sent in previous message.

```
Fields:
private TokenCredential credentials;

Methods:
ServiceBusJmsConnectionFactory connectionString(String connectionString);
ServiceBusJmsConnectionFactory credential(String fullyQualifiedNamespace, TokenCredential credential);
ServiceBusJmsConnectionFactory credential(String fullyQualifiedNamespace, AzureNamedKeyCredential);
ServiceBusJmsConnectionFactory credential(String fullyQualifiedNamespace, AzureSasCredential credential);
```